### ISSUE TITLE
v1.0rc2 breaks on Android

### ISSUE THREAD

**Issue by outerbound** [chrisben/imgcache.js#148]
I upgraded to v1.0rc2. Imgcache is totally broken on Android. I checked logs, Why is cdvfile:// coming for Android? I reverted back to v1.0rc1 and things are back to normal.

rc2 log
--------
INFO: ImgCache initialising
INFO: LocalFileSystem opened
INFO: Local cache folder opened: cdvfile://localhost/persistent/ImgCache/
INFO: .nomedia file created.
cdvfile://localhost/persistent/ImgCache/31ce96647d9ebd5b22332116d74232a03bca3fbd.jpg Failed to load resource: the server responded with a status of 404 (Not Found)
cdvfile://localhost/persistent/ImgCache/a797e4a94395220a55f99135604f4f26779eb353.png Failed to load resource: the server responded with a status of 404 (Not Found)

rc1 log
-----------
INFO: ImgCache initialising
INFO: LocalFileSystem opened
INFO: Local cache folder opened: file:///storage/emulated/0/ImgCache/
INFO: .nomedia file created.

**Comment by chrisben** [comment]
I suppose this is a regression caused by this change: 
https://github.com/chrisben/imgcache.js/commit/9160fb4d6609011e0796d5c124117965f655224f
That seemed to fix things for some people (see #97) 

Could you please:
- state the version of Android you're using
- state the version of Cordova / File plugin
- try to comment lines 177 to 179 of imgcache.js to see if it works without

**Comment by ksibod** [comment]
I'm not sure if @outerbound was able to fix their issue or not, but I was having the same problem.  @chrisben You are correct that commenting out the three lines added in [9160fb4](https://github.com/chrisben/imgcache.js/commit/9160fb4d6609011e0796d5c124117965f655224f) solved the issue. 

This is with:
* Android 5.0.2
* Cordova Android - 5.0.0 
* File Plugin  - 3.0.0

**Comment by chrisben** [comment]
Thanks @ksibod for the feedback!
Perhaps then those lines of code should only be kept for iOS then.. Reading the comments on #97 suggests most people saying it fixed their problems were iOS user.
@ksibod or @outerbound  : would you be able to test your application on iOS ?

**Comment by eekilli** [comment]
Hi, maybe not completely related to this issue, but I was having 404 errors using ImgCache on Android (4.0+) with Cordova 5. After much trial and error I found the solution of adding
```<access origin="cdvfile://*" />``` to my config.xml. (If you use the whitelist plugin you probably have to add ```<allow-intent href="cdvfile://*" />``` as well)

I could not find anything about this in the CORDOVA.md, so would be nice if someone could add this to prevent anyone else having the same headache :)

**Comment by ksibod** [comment]
@chrisben No problem! 
I did test this on iOS as well. For my application, there was no error either way.  Commenting those lines out didn't change anything at all. This was on an iPhone 6 with iOS 9.1.

**Comment by JumpLink** [comment]
Same problem and fixed with @eekilli's solution, thank you!
See also [this similar issue](https://github.com/markmarijnissen/cordova-app-loader/issues/39#issuecomment-136628481).

**Comment by SamVerschueren** [comment]
I confirm adding `<access origin="cdvfile://*" />` to the `config.xml` file fixes this problem.

**Comment by chrisben** [comment]
Great, I'm adding this info to the documentation. Thanks @eekilli

**Comment by elbidone** [comment]
I still observe this issue. I've migrated from imgcache.js#v1.0rc1 to imgcache.js#v1.0.0 and stop working on Android. Works well on iOS. If I do the reverse migration, re-install version 1.0rc1 it works back.
I've tested with phonegap cli-6.0.0 (Android 5.1.0) and File plugin 4.1.0.
My device run with Android 4.1.
On my config.xml I've setted both lines:

```bash
<access origin="cdvfile://*" />
<allow-intent href="cdvfile://*" />
```
 
Can't see errors in logs
```bash
[phonegap] [console.log] INFO: ImgCache initialising
[phonegap] [console.log] INFO: LocalFileSystem opened
[phonegap] [console.log] INFO: Local cache folder opened: cdvfile://localhost/persistent/imgcache/
[phonegap] [console.log] INFO: .nomedia file created.
```

**Comment by chrisben** [comment]
@elbidone would that be possible for you to track when that regression occurred by testing the successive commits following rc1? You can use dichotomy to speed up the process there.
Sorry I make you work there, but I don't have all the environment to reproduce your issue.

**Comment by elbidone** [comment]
Hi @chrisben, I've finally found the issue. 

After days of investigation (I've tested various versions of phonegap and File plugin), the solution comes from a PhoneGap Community Issue: [cdvfile throws 404 not found error in Phonegap 5.1.1 ](http://community.phonegap.com/nitobi/topics/cdvfile-throws-404-not-found-error-in-phonegap-5-1-1?full_site=1).
It only works for me if I declare on my config.xml:

```xml
<access origin="cdvfile://*" />
<allow-navigation href="cdvfile://*" />
```

If I remove `<allow-navigation href="cdvfile://*" />` it raises 404. 

I didn't need to add cdvfile scheme on my CSP or `<allow-intent href="cdvfile://*" />` on my config.xml.
Below my CSP. 

```xml
<meta http-equiv="Content-Security-Policy" content="default-src *; style-src self' 'unsafe-inline'; script-src * 'unsafe-inline' 'unsafe-eval'; img-src * filesystem:;">
```

Thanks for your reply, this issue was an opportunity do dig deeper into imgcache.js code, congratulations it's very well implemented and commented!!
